app/models/health_card.rb

```diff
@@ -0,0 +1,16 @@
+class HealthCard
```

Comment:
This class seems unnecessary and confusing. Having the same `HealthCard` name, when we already have a `HealthCard` class from a library is bound to cause confusion.

This class is only used in `health_card_controller`, but only partially encapsulates the logic there. It seems like this class should also take a `vaccine_group` argument and `filter` within the class.

Even then I'm not sure this class does enough useful things to be its own thing and it would be clearer to understand what's going on to just put this logic in a simple `filter_immunizations` method that `def create` in the controller calls.